distributed/comm/ucx.py

```diff
@@ -0,0 +1,366 @@
+"""
+:ref:`UCX`_ based communications for distributed.
+
+See :ref:`communcations` for more.
+
+.. _UCX: https://github.com/openucx/ucx
+"""
+import asyncio
+import itertools
+import logging
+import struct
+
+from .addressing import parse_host_port, unparse_host_port
+from .core import Comm, Connector, Listener, CommClosedError
+from .registry import Backend, backends
+from .utils import ensure_concrete_host, to_frames, from_frames
+from ..utils import ensure_ip, get_ip, get_ipv6, nbytes
+
+import ucp
+
+import os
+
+os.environ.setdefault("UCX_RNDV_SCHEME", "put_zcopy")
+os.environ.setdefault("UCX_MEMTYPE_CACHE", "n")
+os.environ.setdefault("UCX_TLS", "rc,cuda_copy")
+
+logger = logging.getLogger(__name__)
+MAX_MSG_LOG = 23
+PORT = 13337
+
+# set in ~/.dask/config.yaml
+# or DASK_DISTRIBUTED__COMM__UCXADDRESS
+_PORT_COUNTER = itertools.count(PORT)
+
+_INITIALIZED = False
+
+
+def _ucp_init():
```

Comment:
It occurs to me now that this behavior of the global `_INITIALIZED` to know whether init has been called probably belongs in `ucp.init`.

Reply:
@Akshay-Venkatesh has this happened?  is `ucp.init()` idempotent?

Reply:
Ah, looks like yes: 9d6b99587b4d748f89256d1a9338afb463e27ea4